casper/src/main/scala/coop/rchain/casper/BlockDag.scala

```diff
@@ -8,17 +9,24 @@ import scala.collection.immutable.{HashMap, HashSet}
 final case class BlockDag(idToBlocks: Map[Int, BlockMessage],
                           blockLookup: Map[BlockHash, BlockMessage],
                           childMap: Map[BlockHash, Set[BlockHash]],
-                          latestMessages: Map[Validator, BlockHash],
+                          latestMessages: LatestMessages,
+                          latestMessagesOfLatestMessages: Map[Validator, LatestMessages],
                           currentId: Int,
                           currentSeqNum: Map[Validator, Int])
 
 object BlockDag {
+  type LatestMessages = Map[Validator, BlockHash]
+  object LatestMessages {
+    def empty(): LatestMessages = HashMap.empty[Validator, BlockHash]
```

Comment:
Parentheses are not needed on this method. Typically, you only include the empty argument list if the function has a side-effect. Not a PR blocker, but should fix in a future commit.

Reply:
Will do in next PR